Redesign the app (un)install receiver

SebiderSushi opened the pull request:
Issues covered: closes #94 closes #113 closes #114
PRs included: closes #110
Feature branches included: SebiderSushi:fix_installtime_froyo_catchblock

SebiderSushi commented:
Okay i've tested this once now for Oreo and it did't work right;
Probably something concerning Broadcasts & JobSheduler.
Should also give an option to reenable refreshing onCreate or even onResume.

SebiderSushi commented:
This commit could be applied as is and work correctly if you would be ready to roll back to targeting SDK Version 25 again @ligi.
If there are any implications in that (e.g. Google Play related) or if you just really want to target android 8 then let me know and I'll update my patches to work with oreo.

Otherwise I don't see any reason to target Oreo and since it's majorly breaking app refreshing functionality as it is implemented right now (even on master) so rolling back and merging this would really be better in my opinion. [The lazy solution]

ligi commented:
Unfortunately I need to target 26 now if I want to upload to google play:
https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html

SebiderSushi commented:
Just saw it in Android Lint too .-.

I see three options:
- Release this to F-Droid and push a Version to Google Play that updates in onResume() untli Oreo compliance is implemented
- Release this to F-Droid and skip one release on Google Play until Oreo compliance is implemented
- Implement Oreo compliance right away for everything

ligi commented:
I would really like to let f-droid version and google play not drift apart - this only makes problems in the long run. 
I only see option #3 as the way to go

SebiderSushi commented:
Okay, digging into it.

SebiderSushi commented:
If anyone can, please test this patch so it will prove to be stable enough for merge.

SebiderSushi commented:
I consider this patch complete now. This means it is not ready for merge but should be fully functional and that any changes from this point will only be code cleanup and bug fixes.
Once the testing build has been thoroughly tested and once someone has reviewed my changes, possibly spotting bad practices or code mess, i will go on and polish it up for merge.

SebiderSushi commented:
Another thing: As you can see in the change listing, some changes in my commit result from using the "Reformat Code" action in Android Studio (with default settings). Of course this is out of scope of this commit and will be removed before merge. I would nontheless welcome it if generally all FAST code would be reformatted in a dedicated commit to follow at least some code style. After all, it should make life easier for everyone working with FAST source code.

What do you think @ligi? Is this a good suggestion or do you want to keep all existing code as is and only apply style guidelines to newly contributed code?

SebiderSushi committed:
feat: make Application Details appear on the recents screen

SebiderSushi committed:
fix(AppInfo): infer update time from apk mtime on Froyo and older

nerd190 commented:
Side note... I F*in love this app, and now SebiderSushi too  for all your efforts! :D

SebiderSushi committed:
refactor: move TargetStore.java to the appropriate package directory

SebiderSushi committed:
chore: update repositories for JCenter end-of-life

SebiderSushi committed:
chore: update sdk, build tools & plugins

 - accept IDE suggestions to refactor lint and packagingOptions in app build.gradle
 - replace package in manifest with namespace in build.gradle
   to comply with a deprecation in gradle build tools 7.3.0

SebiderSushi committed:
feat: use https in sharing links

SebiderSushi committed:
chore: add applicationIdSuffix and unique app_name for debug builds

SebiderSushi committed:
chore: take care of no longer relevant TODO items about @TargetApi(5)

SebiderSushi committed:
fix(SortByLastInstalled): change direction to newest first

SebiderSushi committed:
fix(SortByLastInstalled): sort pinned apps to the top

SebiderSushi committed:
fix(LoadingDialog): properly deal with screen rotation instead of locking it

SebiderSushi committed:
chore: specify whether to export activities and receivers

SebiderSushi committed:
refactor: remove unused classes ActivityHelper and ContextHelper

SebiderSushi committed:
refactor(Layout): remove redundant nesting & Reformat Code for Title Bars

SebiderSushi committed:
chore: request permission for uninstalling apps on api 28+

SebiderSushi committed:
chore: request permission for querying apps on api 30+

SebiderSushi committed:
fix(Layout): remove extra padding at the bottom of item_icon.xml

SebiderSushi committed:
fix(AppInfo): track last update time separately

SebiderSushi committed:
fix(translation): add missing array values and improve values-de

SebiderSushi committed:
fix(AppInfo): propagate install time when merging new items

SebiderSushi committed:
chore: use suffixes to generate applicationIds for variants

SebiderSushi committed:
feat: when sharing, use "[app label] - [url]"

SebiderSushi committed:
chore: hide "Open as Notification" on android 8 and above

broken since sdk level 26:
  requires notification channel support
crashes on android 12

SebiderSushi committed:
chore: targetSdkVersion 32

SebiderSushi committed:
fix(AppInfo): silently handle NameNotFoundException when looking up install time

SebiderSushi committed:
refactor(AppInfo): remove getAlternatePackageName

Package IDs may only contain [a-zA-Z0-9_]
https://developer.android.com/studio/build/configure-app-module

SebiderSushi committed:
refactor: remove unused ViewHelper.findById

SebiderSushi committed:
feat: refresh an app icon if its app was updated

SebiderSushi committed:
refactor(AppInfo): remove duplicated accent stripping code

SebiderSushi committed:
feat: strip accents with java.text.Normalizer if available

SebiderSushi committed:
fix(AppInfo): calculate the alternate label from the display label

SebiderSushi committed:
fix(AppInfo): invalidate if activityInfo is null

SebiderSushi committed:
refactor(AppInfo): minor cleanup

SebiderSushi committed:
fix(AppIconCache): remove log spam

SebiderSushi committed:
fix(AppIconCache): close the FileInputStream in getIcon()

SebiderSushi committed:
wip: redesign

SebiderSushi committed:
fix(tests): remove unresolved imports in TheAppInfoList and TheStringUtils

SebiderSushi committed:
refactor(AppInfo): invert boolean isValid

SebiderSushi committed:
refactor(AppInfo): parse CSV with a switch statement

SebiderSushi committed:
fix(Open as Notification): specify the pending intent to me immutable

SebiderSushi committed:
chore: remove redundant suppressions

SebiderSushi committed:
chore: reminders for non-localized strings

SebiderSushi committed:
refactor: remove unused ViewHelper.hideKeyBoard and .startIntentOnClick

SebiderSushi committed:
refactor: remove unused DynamicAppInfoList.get*WithHash

SebiderSushi committed:
refactor(AppInfo): remove setCallCount and getOverrideLabel

SebiderSushi commented:
Branch is supposed to be ready for review now. Feel free to exclude or cherry-pick commits as you see fit.  
Please do not merge this PR until i've had a chance to do a final rebase and squash everything i.e. if there still are commits that start with `fixup` or `wip`

I've put together some [release notes](https://github.com/SebiderSushi/FAST/releases) for this branch for a quick summary.

Remaining issues:
- A lot of Dependencies are deprecated or updating would incur additional work. or raise the `minSdkLevel`

SebiderSushi committed:
fix(strings): be more clear on what "Clear cache" does

SebiderSushi committed:
chore: remove Manifest override for debug builds

Duplicate content roots are no longer supported since
Android Gradle Plugin 7.2.0 / Android Studio Chipmunk 2021.2.1
https://issuetracker.google.com/issues/232007221

Additionally, the referenced file does not even exist anymore.

SebiderSushi committed:
feat: make FAST appear on the recents screen

SebiderSushi committed:
chore(documentation): categorize AppInfo member variables

Variable declarations are sorted into the following categories:
 - Static information that uniquely identifies the activity
 - Dynamic information that can be regenerated and may need updating
 - User generated data that needs to be preserved at all times
 - Runtime state

SebiderSushi committed:
chore(documentation): document AppInfo.labelMode

SebiderSushi committed:
feat: don't dismiss the Change Label dialog when touching outside

SebiderSushi committed:
refactor: remove unused class PackageListSerializer

its code is entirely duplicated in AppInfoListStore

SebiderSushi committed:
fix(strings): be more clear on what "Convert umlauts" does

SebiderSushi committed:
feat: trim the icon cache after refreshing

SebiderSushi committed:
feat: keep index when changing icon resolution

SebiderSushi committed:
feat: detect adaptive icon change
